.github/workflows/pr_label_size.yml

```diff
@@ -0,0 +1,33 @@
+name: labeler
+
+on: [pull_request]
+
+jobs:
+    labeler:
+      permissions:
+        pull-requests: write
+        contents: read
+        issues: write
+      runs-on: ubuntu-latest
+      name: Label the PR size
+      steps:
+        - uses: codelytv/pr-size-labeler@v1
+          with:
+            xs_label: 'size:XS'
+            xs_max_size: '10'
+            s_label: 'size:S'
+            s_max_size: '100'
+            m_label: 'size:M'
+            m_max_size: '500'
+            l_label: 'size:L'
+            l_max_size: '1000'
+            xl_label: 'size:XL'
+            fail_if_xl: 'false'
+            message_if_xl: >
+                This PR exceeds the recommended size of 1000 lines.
+                Please make sure you are NOT addressing multiple issues with one PR.
+                Note this PR might be rejected due to its size.
+            github_api_url: 'https://api.github.com'
+            files_to_ignore: '*.md .gitignore'
```

Comment:
Maybe as a tech debt, we should also ignore auto-generated files (minor changes in `parser.y` can cause a big diff in `parser.c`, for example)